### ISSUE TITLE
Dashboard Transaction Log Page

### ISSUE THREAD

**Issue by smombartz** [unlock-protocol/unlock#526]
## Description
A separate page that shows all actions and transactions – essentially a log. Currently accounted for are:
- Creating a new Lock
- Renaming a Lock
- Lock/Key price change
- Key purchases
- Withdrawing funds
- Downloading data

Also note the new Log Page icon in the header of the dashboard (see screenshot below).

Specs: https://zpl.io/2vNZKOn
Assets: [log.svg](https://github.com/unlock-protocol/unlock/blob/master/unlock-app/src/static/images/svg/log.svg)

## Screenshot
![dashboard transaction log](https://user-images.githubusercontent.com/1377389/48371483-047d5c80-e68a-11e8-81d7-9f904fa59601.png)

<img width="922" alt="log icon" src="https://user-images.githubusercontent.com/1377389/48367343-5e782500-e67e-11e8-9756-20cc25e98a23.png">

**Comment by smombartz** [comment]
good one! forgot to mention and just updated the issue. there is an icon in the header to get to the page. getting back to dashboard I was thinking either click on log icon or on address. what you think?

**Comment by benwerd** [comment]
Aha! That makes sense. Maybe the log icon could be highlighted while we're looking at the log to make it clear that it can be untoggled? Or something like that?

**Comment by julien51** [comment]
@benwerd Good point. I think generally we should also hightlight the jobs/about icons when on the corresponding pages!

**Comment by julien51** [comment]
@benwerd Removing this from your place because I believe this can be a great application contribution!

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 2.002 ETH (221.99 USD @ $110.88/ETH)  attached to it as part of the Unlock Inc fund.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $33,985.35 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by julien51** [comment]
Please do not start to work on this unless you have been previously approved by our team to work on this.

We reserve some of our issue for applicants to our full time positions. Thanks for your understanding.

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Workers have applied to start work__.


These users each claimed they can complete the work by 1 week, 6 days from now.
Please review their action plans below:


**1) [njmyers](https://gitcoin.co/profile/njmyers) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?mutate_worker_action=approve&worker=njmyers) | [reject worker](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?mutate_worker_action=reject&worker=njmyers))_.**

Happy to start working on this feature. Thanks Julien!

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846).

**Comment by njmyers** [comment]
@julien51 Hi there! I've started work on the transaction log page. Do you have any methods that you are using to simulate transactions? It appears I would need some sample data and I wasn't able to find any code that clearly defined the object shape of a transaction. Thanks for any advice!

**Comment by julien51** [comment]
Sorry for the delay... So, you can find how transactions are created in the `web3Service` file.
I would suggest creating a couple locks locally (which would create transactions!).

**Comment by julien51** [comment]
@njmyers What's the status on that one? Can I help in any way?

**Comment by njmyers** [comment]
Hi @julien51. Yes I have a few questions.

1) Is the intention to have the log page at a new url like /dashboard/log? I was playing around with creating new pages using the next interface but there seems to be some issues especially when navigating to the dashboard directly. When I startup the app and navigate to http://localhost:3000/dashboard I get an error that is preventing me from creating the logs page on a separate url. Maybe it is just related to my local setup.

2) I am able to get transactions from test locks but they don't have any amounts! I really searched around and am trying to learn about ethereum network but I cannot find a consistent way to simulate actual transactions for testing purposes. The new lock transactions have this shape...

`
{
status":"mined",
"confirmations":255,"
createdAt":1544027371730,
"hash":"0xf094c11b49dcd71d2c787cd98a36e846a9c57c1d85b79e6d759dbeae817007bb"
}
`

It just seems to me that this does not have enough relevant information for the log purposes as is described in the specs. 

Any information or resources would be appreciated thanks Julien!

**Comment by julien51** [comment]
Thanks!

1. I think maybe just `/log` would be good actually. I think you need to create a new "Page" component similar to the Dashboard component and then make sure you link to it inside the next config?

2. We build our own transaction objects rather than re-use the ones provided by the web3 library as this will let us change libraries more easily if/when we need. All of the `transaction` objects are created inside the `web3Service` library on function calls which do need them (`createLock` , `purchaseKey`, `withdraw`....). I think you can easily get the amount of these in there ;)

Let me know if that helps!

**Comment by njmyers** [comment]
@julien51 Yes that is most helpful. Yes I was able to create the log page but for some reason putting it under /dashboard/log was causing an issue. I think it is possibly related to #667 and the SSR component that you are using. I'll simply work with /log for now.

Thanks for your explanations regarding the objects. That makes sense but since I am a ethereum newbie it had me thoroughly confused!

**Comment by gitcoinbot** [comment]
@njmyers Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] reminder (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=1) | [3 days](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=3) | [5 days](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=5) | [10 days](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=10) | [100 days](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=100)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

@njmyers due to inactivity, we have escalated [this issue](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846) to Gitcoin's moderation team. Let us know if you believe this has been done in error!

* [x] reminder (3 days)
* [x] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=1) | [3 days](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=3) | [5 days](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=5) | [10 days](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=10) | [100 days](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=100)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

@njmyers due to inactivity, we have escalated [this issue](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846) to Gitcoin's moderation team. Let us know if you believe this has been done in error!

* [x] reminder (3 days)
* [x] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=1) | [3 days](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=3) | [5 days](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=5) | [10 days](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=10) | [100 days](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=100)

**Comment by gitcoinbot** [comment]
@njmyers Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] reminder (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=1) | [3 days](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=3) | [5 days](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=5) | [10 days](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=10) | [100 days](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=100)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

@njmyers due to inactivity, we have escalated [this issue](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846) to Gitcoin's moderation team. Let us know if you believe this has been done in error!

* [x] reminder (3 days)
* [x] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=1) | [3 days](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=3) | [5 days](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=5) | [10 days](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=10) | [100 days](https://gitcoin.co/issue/unlock-protocol/unlock/526/1846?snooze=100)

**Comment by julien51** [comment]
I think @cnasc is almost done with it!

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Cancelled** 

<hr>

__The funding of 2.002 ETH  (294.42 USD @ $147.06/ETH) attached to this issue has been **cancelled** by the bounty submitter__

 
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $59,822.46 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by cnasc** [comment]
Yes, didn't think to check to see if there was an older issue than from our sprint. At this point an early iteration of the transaction log is in place, though there is no navigation linking to it.

**Comment by cnasc** [comment]
At this point the log page is implemented, and I think future work on that component will come under a separate heading so we can close this.
